apis/condition_set.go

```diff
@@ -295,6 +295,7 @@ func (r conditionsImpl) recomputeHappiness(t ConditionType) {
 		r.SetCondition(Condition{
 			Type:     r.happy,
 			Status:   corev1.ConditionTrue,
+			Reason: reason,
```

Comment:
normally the happy condition does not have a reason. The reason it is happy is because _everything_ else is happy. It is not a lot of signal to propagate the last flop of ready=true from a dependent to happy.

Reply:
OK!   Is it means that only when happy condition's status is false, there will need a reason?  Otherwise, no reason will need?

Reply:
Is there any document or contract about how to use conditions like when to set reason or not.  Is it just convention?

Thank you very much !

Reply:
> OK! Is it means that only when happy condition's status is false, there will need a reason? Otherwise, no reason will need?

Yeah - to Scott's point there really isn't an need to propagate a reason why the `Ready` Condition is happy - it should be inferred from the dependent child conditions

> Is there any document or contract about how to use conditions like when to set reason or not? Is it just convention?

Usually the Type is descriptive enough so you wound't need an additional reason string when the condition is `True`. 

`Reason` & `Message` are for situations when a condition cannot be satisfied - ie. Status=`False` and we want to convey additional information